integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala

```diff
@@ -120,9 +120,16 @@ object CarbonStore {
               (-1L, -1L)
             }
           } else {
-            // for batch segment, we can get the data size from table status file directly
-            (if (load.getDataSize == null) -1L else load.getDataSize.toLong,
-              if (load.getIndexSize == null) -1L else load.getIndexSize.toLong)
+            // If the added segment is other than carbon segment then we can only display the
+            // data size and not index size and for batch segment, we can get the data size
+            // from table status file directly
+            if (!load.getFileFormat.equals(FileFormat.COLUMNAR_V3) &&
+                !load.getFileFormat.equals(FileFormat.ROW_V1)) {
+              (if (load.getIndexSize == null) -1L else load.getIndexSize.toLong, -1L)
```

Comment:
but here you are getting index size also from table status file, and not getting the data size

Reply:
and please add testcase to check data size and index size by `show segment`

Reply:
The index size is calculated as the size of the files present in the location, so here it gives the correct size of the corresponding parquet/orc files.

Reply:
please add a testcase to check the data size

Reply:
Done